### PR TITLE
Closes #78: Generate TODO.md from @todo (and similar) comments

### LINKED ISSUE #78: Generate list of todo and hack comments
@Todo tags
@Hack tags 
and others should be found and outputted into a .md file.

### PR DESCRIPTION
**Pull request by Bugvi-Benjamin-M**
Started over due to the other pull request regarding #78 was bloated, 6 months old and on life support.

### CODE REVIEW

File: scripts/helpers/todos.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+ * @param {Iterable<Todo>} todos
+ */
+function writeTodos(file, todos) {
+  /**
+   * @type {Map<string, Array<string>>}
+   */
+  const groups = new Map();
+
+  for (let { file, comment } of todos) {
+    const lines = parseLines(readFile(file));
```
Comment (kasperisager) [inline scripts/helpers/todos.js]: See: https://github.com/Siteimprove/alfa/pull/125#discussion_r294247347

File: scripts/helpers/todos.js [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+ * @return {Iterable<Todo>}
+ */
+function getTodos(file, project) {
+  return [...project.getTodos(file, checks)].map(comment => ({
+    file,
+    comment
+  }));
+}
+
+exports.getTodos = getTodos;
```
Comment (NiclasHedam) [inline scripts/helpers/todos.js]: I think you can just write `export function getTodos(file, project) { ... `

File: scripts/helpers/graph.js [hunk 3]
```diff
@@ -46,6 +46,10 @@ class Graph {
       for (const neighbour of edges.incoming) {
         this.removeEdge(neighbour, node);
```
Comment (NiclasHedam) [inline scripts/helpers/graph.js]: Is `this` necessary?

File: yarn.lock [hunk 4]
```diff
@@ -447,9 +447,9 @@
   integrity sha512-yALhelO3i0hqZwhjtcr6dYyaLoCHbAMshwtj6cGxTvHZAKXHsYGdff6E8EPw3xLKY0ELUTQ69Q1rQiJENnccMA==
 
 "@types/jest@^24.0.13", "@types/jest@^24.0.6":
-  version "24.0.14"
```
Comment (NiclasHedam) [inline yarn.lock]: Don't update the dependencies. Restore the original content of yarn.lock.

File: scripts/helpers/todos.js (hunk 2 above)
Comment (kasperisager) [reply scripts/helpers/todos.js]: The module support in Node.js is both experimental and requires use of the `.mjs` file extension so all files in `scripts/` must use CJS.

File: scripts/helpers/graph.js (hunk 3 above)
Comment (kasperisager) [reply scripts/helpers/graph.js]: Yes, `removeEdge` is a class method.

File: yarn.lock (hunk 4 above)
Comment (kasperisager) [reply yarn.lock]: Seems like a merge gone wrong; needs a rebase (@Bugvi-Benjamin-M).

File: scripts/helpers/todos.js [hunk 5]
```diff
@@ -76,7 +76,7 @@ function writeTodos(file, todos) {
     content += `# ${item.pkg}\n`;
 
     for (const [type, todos] of item.contentByTypeGrouped) {
-      content += `## ${type}\n`;
+      content += `### ${type}\n`;
```
Comment (kasperisager) [inline scripts/helpers/todos.js]: Headings should be logically nested and no levels skipped (https://www.w3.org/WAI/WCAG21/Techniques/general/G141.html).

**Pull request merged**